packages/block-library/src/rich-image/rich-image/index.js

```diff
@@ -0,0 +1,357 @@
+/**
+ * External dependencies
+ */
+
+import classnames from 'classnames';
+import Cropper from 'react-easy-crop';
+
+/**
+ * WordPress dependencies
+ */
+
+import {
+	BlockControls,
+	__experimentalBlock as Block,
+} from '@wordpress/block-editor';
+import { Fragment, Component } from '@wordpress/element';
+import {
+	Toolbar,
+	ToolbarButton,
+	Icon,
+	Button,
+	Spinner,
+	withNotices,
+	RangeControl,
+	DropdownMenu,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { compose } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import richImageRequest from './api';
+import {
+	RotateLeftIcon,
+	RotateRightIcon,
+	FlipHorizontalIcon,
+	FlipVerticalIcon,
+	CropIcon,
+	AspectIcon,
+} from './icon';
+
+const ROTATE_STEP = 90;
+const DEFAULT_CROP = {
+	unit: '%',
+	x: 25,
+	y: 25,
+	width: 50,
+	height: 50,
+};
+const MIN_ZOOM = 1;
+const MAX_ZOOM = 3;
+const ZOOM_STEP = 0.1;
+const POPOVER_PROPS = { position: 'bottom right' };
+
+class RichImage extends Component {
```

Comment:
It would be nice to create new components with hooks.

Reply:
In fact, it is recommended by our [guidelines](https://developer.wordpress.org/block-editor/contributors/develop/coding-guidelines/#documenting-wordpresselement-react-components):

> When possible, all components should be implemented as function components, using hooks for managing component lifecycle and state.

Reply:
Agreed. And, again, this is mostly carried over from Automattic/block-experiments. It'll get mostly rewritten as part of #22581.